index.js

```diff
@@ -1,149 +1,136 @@
+    if (is.existy(req.connection)) {
+        if (is.ip(req.connection.remoteAddress)) {
+            return req.connection.remoteAddress;
+        }
+        if (is.existy(req.connection.socket) && is.ip(req.connection.socket.remoteAddress)) {
+            return req.connection.socket.remoteAddress;
+        }
     }
-    if ((ipAddress = req.info && req.info.remoteAddress)) {
-        return ipAddress;
-    } 
-
-    // return null if we cannot find an address
-    return null;
-}
 
-function parseForwardedForAlt(forwardedForAlt) {
-    if (!forwardedForAlt) {
-        return;
+    if (is.existy(req.socket) && is.ip(req.socket.remoteAddress)) {
+        return req.socket.remoteAddress;
     }
-    // x-forwarded-for may return multiple IP addresses in the format:
-    // "client IP, proxy 1 IP, proxy 2 IP"
-    // Therefore, the right-most IP address is the IP address of the most recent proxy
-    // and the left-most IP address is the IP address of the originating client.
-    // source: http://docs.aws.amazon.com/elasticloadbalancing/latest/classic/x-forwarded-headers.html
-    var forwardedIps = forwardedForAlt.split(',');
-    // Sometimes IP addresses in this header can be 'unknown' (http://stackoverflow.com/a/11285650).
-    // Therefore taking the left-most IP address that is not unknown
-    // A Squid configuration directive can also set the value to "unknown" (http://www.squid-cache.org/Doc/config/forwarded_for/)
-    for (var i = 0; i < forwardedIps.length; ++i) {
-        var ipAddress = forwardedIps[i].trim();
-        if (ipAddress && ipAddress !== 'unknown') {
-            return ipAddress;
-        }
+
+    if (is.existy(req.info) && is.ip(req.info.remoteAddress)) {
+        return req.info.remoteAddress;
     }
-}
 
+    return '';
+}
 
 /**
- * Expose mode public functions
+ * Expose request IP as a middleware.
+ *
+ * @param {object} [options] - Configuration.
+ * @param {string} [options.attributeName] - Name of attribute to augment request object with.
+ * @return {*}
  */
-exports.getClientIp = getClientIp;
+function mw(options) {
+    // Defaults.
+    const configuration = is.not.existy(options) ? {} : options;
 
+    // Validation.
+    if (is.not.object(configuration)) {
+        throw new TypeError('Options must be an object!');
+    }
 
-/**
- * Expose a default implemtation for a connect middleware
- * 
- * @options.attributeName - name of attribute to augment request object with
- */
-exports.mw = function(options) {
-    if (!options) options = {};
-    var attr = options.attributeName || "clientIp";
-    return function(req, res, next) {
-        req[attr] = getClientIp(req);
+    const attributeName = configuration.attributeName || 'clientIp';
+    return (req, res, next) => {
+        req[attributeName] = getClientIp(req); // eslint-disable-line no-param-reassign
         next();
-    }
-};
\ No newline at end of file
+    };
+}
+
+module.exports = {
+    getClientIpFromXForwardedFor,
```

Comment:
I don't think we need to export `getClientIpFromXForwardedFor`. Should be a private method for `getClientIp`.